packages/opencensus-web-instrumentation-zone/test/test-perf-resource-timing-selector.ts

```diff
@@ -0,0 +1,216 @@
+      );
+      expect(filteredPerfEntries.length).toBe(3);
+      expect(filteredPerfEntries).toContain(entry1);
+      expect(filteredPerfEntries).toContain(entry2);
+      expect(filteredPerfEntries).toContain(entry3);
+    });
+
+    it('Should take single and tuples of resource timing entries', () => {
+      const entry1 = createFakePerfResourceEntry(
+        2.100000011036173,
+        5.500000011036173
+      );
+      const entry2 = createFakePerfResourceEntry(
+        6.500000011036173,
+        10.200000011036173
+      );
+      const entry3 = createFakePerfResourceEntry(
+        8.500000011036173,
+        20.000000000000003
+      );
+      const entry4 = createFakePerfResourceEntry(
+        1.500000011036173,
+        25.000000000000003
+      );
+      const perfResourceEntries = [entry1, entry2, entry3, entry4];
+      const filteredPerfEntries = getPossiblePerfResourceEntries(
+        perfResourceEntries
+      );
+      expect(filteredPerfEntries.length).toBe(3);
+      expect(filteredPerfEntries).toContain([entry1, entry2]);
+      expect(filteredPerfEntries).toContain([entry1, entry3]);
+      expect(filteredPerfEntries).not.toContain([entry1, entry4]);
+      expect(filteredPerfEntries).not.toContain([entry2, entry3]);
+      expect(filteredPerfEntries).not.toContain([entry2, entry4]);
+      // Should not contain the single resource entries as they already were
+      // paired.
+      expect(filteredPerfEntries).not.toContain(entry1);
+      expect(filteredPerfEntries).not.toContain(entry2);
+      expect(filteredPerfEntries).not.toContain(entry3);
+      // As entry4 has not been paired, should contain this a single entry.
+      expect(filteredPerfEntries).toContain(entry4);
+    });
+  });
+
+  describe('getXhrPerfomanceData', () => {
+    it('Should take the tuple with the minimum gap to the span as the best resource timing entry', () => {
+      const entry1 = createFakePerfResourceEntry(
+        13.100000011036173,
+        15.100000011036173
+      );
+      const entry2 = createFakePerfResourceEntry(
+        13.200000011036173,
+        15.200000011036173
+      );
+      const entry3 = createFakePerfResourceEntry(
+        16.100000011036173,
+        18.100000011036173
+      );
+      const entry4 = createFakePerfResourceEntry(
+        16.200000011036173,
+        18.200000011036173
+      );
+      // Ignore this as its out of the span end time.
+      const entry5 = createFakePerfResourceEntry(
+        35.100000011036173,
+        40.100000011036173
+      );
+      const perfResourceEntries = [entry1, entry2, entry3, entry4, entry5];
+      spyPerfEntryByType(perfResourceEntries);
+      const span = createSpan(13.1, 18.3);
+
+      const filteredPerfEntries = getXhrPerfomanceData('/test', span);
+      expect(filteredPerfEntries).toEqual([entry1, entry4]);
+    });
+
+    it('Should take the single resource entry with the minimum gap to the span as the best entry', () => {
```

Comment:
How is this case different from the one above?

Reply:
This test case is different in the sense that the returned value does not have a cors preflight request value as this is another possible return value. I made this more clear.